CloudBees CI failed due to unrecognized param -Xdoclint:none for maven-javadoc-plugin

CloudBees CI uses JDK1.7 and it doesn't recognize `-Xdoclint:none`

    [ERROR] Exit code: 1 - javadoc: error - invalid flag: -Xdoclint:none
    [ERROR] Command line was: /opt/java/sdk/Oracle/x86_64/jdk1.7.0_79/jre/../bin/javadoc @options @packages

Need figure out a way to cater for JDK8 and earlier versions at the same time.
https://jenkins.ci.cloudbees.com/job/plugins/job/coordinator-plugin/31/console

--- a/pom.xml
+++ b/pom.xml
@@ -115,19 +115,12 @@
 			</plugin>
 			<plugin>
 				<artifactId>maven-javadoc-plugin</artifactId>
-				<!-- 
-					fix failure on jdk8  
-					according to http://stackoverflow.com/questions/15886209/maven-is-not-working-in-java-8-when-javadoc-tags-are-incomplete
-				-->
 				<executions>
 					<execution>
 						<id>attach-javadocs</id>
 						<goals>
-						<goal>jar</goal>
+							<goal>jar</goal>
 						</goals>
-						<configuration>
-						<additionalparam>-Xdoclint:none</additionalparam>
-						</configuration>
 					</execution>
 				</executions>
 			</plugin>
@@ -137,4 +130,28 @@
 			</plugin>
 		</plugins>
 	</build>
+	<profiles>
+		<profile>
+		    <id>doclint-java8-disable</id>
+		    <activation>
+				<jdk>[1.8,)</jdk>
+			</activation>
+		    <build>
+		    	<plugins>
+					<plugin>
+					    <artifactId>maven-javadoc-plugin</artifactId>
+					    <configuration>
+					    	<!-- 
+								fix failure on jdk8  
+								according to http://stackoverflow.com/questions/15886209/maven-is-not-working-in-java-8-when-javadoc-tags-are-incomplete
+								
+								fix #21, CloudBees CI failed due to unrecognized param -Xdoclint:none for maven-javadoc-plugin
+							-->
+					    	<additionalparam>-Xdoclint:none</additionalparam>
+					    </configuration>
+					</plugin>
+				</plugins>
+			</build>
+		</profile>
+	</profiles>
 </project>

